test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs

```diff
@@ -0,0 +1,155 @@
+            Assert.Null(project.EntryPoint);
+        }
+
+        [Fact]
+        public void GetsProjectInfo_FromPathHavingProjectJsonFile()
+        {
+            // Arrange
+            var projectJsonPath = Path.Combine(_testProjectPath, "project.json");
+
+            // Act & Assert
+            Project project;
+            Assert.True(Project.TryGetProject(projectJsonPath, out project));
+            Assert.NotNull(project);
+            Assert.Equal(projectName, project.Name);
+            Assert.NotNull(project.Commands);
+            Assert.Equal(1, project.Commands.Count);
+            string val;
+            Assert.True(project.Commands.TryGetValue("test", out val));
+            Assert.Equal("xunit.runner.aspnet", val);
+            Assert.Null(project.EntryPoint);
+        }
+
+        [Fact]
+        public void GetsProjectInfo_FromRelativeProjectDirectoryPath()
+        {
+            // Arrange
+            var projectDirPath = $"../{projectName}";
+
+            // Act & Assert
+            Project project;
+            Assert.True(Project.TryGetProject(projectDirPath, out project));
+            Assert.NotNull(project);
+            Assert.Equal(projectName, project.Name);
+            Assert.NotNull(project.Commands);
+            Assert.Equal(1, project.Commands.Count);
+            string val;
+            Assert.True(project.Commands.TryGetValue("test", out val));
+            Assert.Equal("xunit.runner.aspnet", val);
+            Assert.Null(project.EntryPoint);
+        }
+
+        [Fact]
+        public void GetsProjectInfo_FromRelativePathHavingProjectJsonFile()
+        {
+            // Arrange
+            var projectDirPath = $"../{projectName}/project.json";
+
+            // Act & Assert
+            Project project;
+            Assert.True(Project.TryGetProject(projectDirPath, out project));
+            Assert.NotNull(project);
+            Assert.Equal(projectName, project.Name);
+            Assert.NotNull(project.Commands);
+            Assert.Equal(1, project.Commands.Count);
+            string val;
+            Assert.True(project.Commands.TryGetValue("test", out val));
+            Assert.Equal("xunit.runner.aspnet", val);
+            Assert.Null(project.EntryPoint);
+        }
+
+        [Fact]
+        public void ThrowsExceptionOnInvalidJsonContent()
+        {
+            // Arrange
+            var projectJsonPath = Path.Combine(_testProjectPath, "project.json");
+            var projectJsonContentStream = new MemoryStream(Encoding.UTF8.GetBytes("not a valid json content"));
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() =>
+                Project.GetProject(projectJsonContentStream, projectJsonPath, projectName));
+            Assert.Equal(
+                $"The JSON file '{projectJsonPath}' can't be deserialized to a JSON object.", exception.Message);
+        }
+
+        [Fact]
+        public void CommandsSetIsEmptyByDefault()
```

Comment:
Some of these tests were borrowed from here: https://github.com/aspnet/dnx/blob/dev/test/Microsoft.Dnx.Runtime.Tests/ProjectFacts.cs